web/src/frontend/classic/stores/Documentation.ts

```diff
@@ -0,0 +1,80 @@
+import { writable } from 'svelte/store';
+
+export type DocCategory = {
+    name: string;
+    description: HTMLElement;
+    sections: DocSection[];
+}
+export type DocSection = {
+    name: string,
+    content: HTMLElement
+}
+
+export const documentation = writable<DocCategory[]>([]);
+
+/**
+ * Creates a slide section by extracting HTML elements from the given content element until a specified split element is encountered.
+ * @param content - The content element from which to extract the slide section.
+ * @param split - The element name at which to stop extracting and split the slide section.
+ * @returns The created slide section element.
+ */
+function slideElement(content:HTMLElement, split:string[]) : HTMLElement {
+    const slide = document.createElement("div");
+    let nextSibling = content.nextSibling;
+    while (nextSibling && !split.includes(nextSibling.nodeName)) {
+        slide.appendChild(nextSibling.cloneNode(true));
+        nextSibling = nextSibling.nextSibling;
+    }
+    return slide;
+}
+
+(async function () {
```

Comment:
Parsing relies heavily on HTML structure. When the side generator is changed, the structure may not be as expected and parsing may fail or the page DOM tree may be composed dynamically.
It would be more reliable to parse the source markdown.

Reply:
the structure is based on simple level2 and level3 makdown and therefore just basic h2 and h3. This wouldn't change
https://raw.githubusercontent.com/manga-download/manga-download.github.io/master/_docs/06-HaruNeko.md
But yes. the markdown could be processed client side (we need to add an extra lib then).

Reply:
should i use a svelte lib or a more generic javascript lib so that it can be used in other interfaces ?
https://www.npmjs.com/package/svelte-markdown